### ISSUE TITLE
NPC's unable to follow in-range checklist for melee [Applicable to Lurker]

### ISSUE THREAD

**Issue by ihcTactics** [Looking4Group/L4G_Core#2895]
NPC's that should be following 2 threatlists/checking for targets in melee range ignore that feature and only look at the threatlist.

If main tank dies and second on threat is someone outside of melee range, even if there is another player in melee range the NPC will act as if no one is.

On Lurker, this leads to him spamming Water Bolt.

I don't know coding so I can't offer a possible fix, but looking into something like a melee range check may be what could fix it.

**Comment by honeyhoney** [comment]
Do you have any examples other than Lurker?

**Comment by ihcTactics** [comment]
I do not. I brought it up as lurker to Anonx and he said it was a general mechanic that wasn't scripted and to list it as either.

That being said, I feel like Murmer has acted in a similar manner but it's been a while since I last did that fight and had a tank die.

**Comment by Anteara** [comment]
Hi, I will take a look at this, if confirmed, it is a very important bug to fix.

**Comment by AnonXS** [comment]
http://wowwiki.wikia.com/wiki/Root?oldid=1610378

A rooted mob that has no ranged attacks will be unable to attack its current aggro target. However, if players are standing in melee range, it will attack the player that has the highest threat level. Thus, it is not wise to stand near a rooted mob. Also, it is not wise to root a mob next to a player that has some threat on it, as it might get him killed.

**Comment by Pbeitelmann** [comment]
I agree with this regarding normal mobs being rooted. I don't agree regarding Bosses;

Lurker shooting waterbolts on the current threat holder, if he is not within melee range to me is intended behaviour. 
If you disagree please provide sources.


Same for Lady Vashj; she should use her bow to range attack the current threat holder, as long as he is not in melee range. She will not switch to a melee target once somebody (that is not current threat holder) is in range.

**Comment by AnonXS** [comment]
https://github.com/cmangos/issues/issues/1378
